core/src/main/java/org/infinispan/persistence/support/NonBlockingStoreAdapter.java

```diff
@@ -41,8 +41,7 @@
 
 public class NonBlockingStoreAdapter<K, V> implements NonBlockingStore<K, V> {
    private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
-   private static final boolean trace = log.isTraceEnabled();
-   private static final AtomicInteger id = trace ? new AtomicInteger() : null;
+   private final AtomicInteger id = log.isTraceEnabled() ? new AtomicInteger() : null;
```

Comment:
IMO we should always create the `AtomicInteger`, and we should only check `log.isTraceEnabled()` in `nextTraceId()`.